app/controllers/todos_controller.rb

```diff
@@ -0,0 +1,56 @@
+class TodosController < ApplicationController
+  def index
+    @incomplete_todos = Todo.where(completed: false).order(due_date: :asc)
+    @complete_todos = Todo.where(completed: true).order(due_date: :asc)
+  end
+
+  def show
+    @todo = Todo.find(params[:id])
+  end
+
+  def new
+    @todo = Todo.new
+  end
+
+  def create
+    @todo = Todo.new(todo_params)
+    if @todo.save
+      flash[:notice] = "You are getting baguettes" # todo fix this
+      redirect_to todos_path
+    else
+      render "new"
+    end
+  end
+
+  def edit
+    @todo = Todo.find(params[:id])
+  end
+
+  def update
+    @todo = Todo.find(params[:id])
+    if @todo.update(todo_params)
+      flash[:notice] = "You added jelly to your baguette"
```

Comment:
I also enjoyed the flash notices. It would have been nice to have different messages when updating a todo and when completing a todo.